src/ops.rs

```diff
@@ -0,0 +1,228 @@
+use std::ops::{Add, Neg, Sub};
+
+use time::{Duration, OffsetDateTime};
+
+use crate::CalendarDuration;
+
+fn add_years(mut time: OffsetDateTime, years: i32) -> OffsetDateTime {
+    loop {
+        match time.replace_year(time.year() + years) {
+            Err(_) => {
```

Comment:
The other copies of this pattern `assert_ne!(time.day(), 1);` to make sure changes to the time crate can't make us loop forever. Should do that here too?

Reply:
sgtm